metrics/registry.go

```diff
@@ -95,18 +93,15 @@ func (r *StandardRegistry) Get(name string) interface{} {
 // alternative to calling Get and Register on failure.
 // The interface can be the metric to register if not found in registry,
 // or a function returning the metric for lazy instantiation.
-func (r *StandardRegistry) GetOrRegister(name string, i interface{}) interface{} {
+func (r *StandardRegistry) GetOrRegister(name string, ctor func() interface{}) interface{} {
 	// fast path
 	cached, ok := r.metrics.Load(name)
 	if ok {
 		return cached
 	}
-	if v := reflect.ValueOf(i); v.Kind() == reflect.Func {
-		i = v.Call(nil)[0].Interface()
-	}
-	item, _, ok := r.loadOrRegister(name, i)
+	item, _, ok := r.loadOrRegister(name, ctor())
 	if !ok {
-		return i
+		return item
```

Comment:
I don't understand this tbh, we return the item regardless of whether loadOrRegister was successful?
Shouldn't we just return nil here?

Reply:
(or ctor())

Reply:
Ah, I missed adding the last change here. We can just return `item`. loadOrRegister sets it to `nil` in case of failure anyways.